service/src/main/kotlin/io/provenance/explorer/service/async/AsyncCaching.kt

```diff
@@ -199,7 +212,8 @@ class AsyncCaching(
                         }
                     }
                 }
-                TxMessageRecord.insert(tx.txResponse.height.toInt(), tx.txResponse.txhash, txId, msg, type, module)
+                val msgId = TxMessageRecord.insert(tx.txResponse.height.toInt(), tx.txResponse.txhash, txId, msg, type, module)
+                saveEvents(txId, tx, msg, msgId, type)
```

Comment:
Youll need to use the index `idx` to match the msg to the proper logsList index. Msg index == logs list index (for now, youll have to do  testing to ensure this lines up). You wont need the msg proper, just the msgId you get back from the save.